libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -113,24 +120,13 @@ func (c *publishClient) Publish(batch publisher.Batch) error {
 			}
 		}
 
-		meta := common.MapStr{
-			"_index":   "",
-			"_routing": nil,
-		}
-		if c.es.GetVersion().Major < 7 {
-			meta["_type"] = t
-		}
-		bulk := [2]interface{}{
-			common.MapStr{"index": meta},
-			report.Event{
-				Timestamp: event.Content.Timestamp,
-				Fields:    event.Content.Fields,
-			},
+		switch c.format {
+		case report.ReportingFormatXPackMonitoringBulk:
+			err = c.publishWithXPackMonitoringBulk(params, event, t)
+		case report.ReportingFormatBulk:
+			err = c.publishWithBulk(event)
```

Comment:
Could we find shorter but descriptive names? e.g. `publishBulk` and `publishXpackBulk` ?